test/python/test_qobj.py

```diff
@@ -42,48 +51,29 @@ def setUp(self):
             ],
         }
 
-    def tearDown(self):
-        self.valid_qobj = None
-        self.expected_dict = None
-
-    def test_create_qobj_and_req_args(self):
-        """Test creation of a Qobj and check for its required args."""
-        self.assertTrue(
-            all(getattr(self.valid_qobj, required_arg) is not None
-                for required_arg in Qobj.REQUIRED_ARGS))
-
-        with self.assertRaises(TypeError):
-            # pylint: disable=no-value-for-parameter
-            Qobj()
-
-    def test_as_dict(self):
-        """Test conversion to dict of a Qobj based on the individual elements."""
-        self.valid_qobj._version = '67890'  # private member variables shouldn't appear in the dict
-        self.assertEqual(self.valid_qobj.as_dict(), self.expected_dict)
+        self.assertEqual(self._valid_qobj.as_dict(), expected_dict)
 
-    def test_as_dict_to_json(self):
+    def test_as_dict_against_schema(self):
         """Test dictionary representation of Qobj against its schema."""
         file_path = self._get_resource_path('qobj_schema.json', Path.SCHEMAS)
 
         with open(file_path, 'r') as schema_file:
             schema = json.load(schema_file)
 
-        jsonschema.validate(self.valid_qobj.as_dict(), schema)
+        passed = False
```

Comment:
Remember when we talked about how `validate()` worked? We assumed it was returning a boolean indicating if the validation was successful or not but it turns out it raises an exception! Your code is OK but you are reimplementing some Python `unittest` functionality such as printing the reason for which the test is failing. You can simply do:

```python
try:
  jsonschema.validate(self._valid_qobj.as_dict(), schema)
except jsonschema.ValidationError as validation_error:
  self.fail(str(validation_error))
```

This is slightly different to the first version of your test where you have the `validate()` line only. If this version enters the `except` block it will **fail**. With your first version, if the line throws, the test result is **error**.